TopMovies/DTO/GraphQL/GraphQLAllCategoriesDTO.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  GraphQLAllCategoriesDTO.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 01.04.2021.
+//
+
+import Foundation
+
+struct GraphQLAllCategoriesDTO: Decodable {
+  let data: DataAllCategoriesDTO
+  struct DataAllCategoriesDTO: Decodable {
+    let movies: MoviesAllCategoriesDTO
+    struct MoviesAllCategoriesDTO: Decodable {
+      let nowPlaying, upcoming, topRated, popular:
+        GraphQLCategoryDTO.DataCategoryDTO.MoviesCategoryDTO.EdgesCategoryDTO
+    }
+  }
+}
+
+extension CategoryDTOWrapper {
+  init(id: String, title: String, dto: GraphQLAllCategoriesDTO, category: MovieCategoryRequest) {
+    self.id = id
+    self.title = title
+    
+    switch category {
+    case .nowPlaying:
+      movies = dto.data.movies.nowPlaying.edges.map { MoviePreviewDTOWrapper.init(dto: $0.node) }
```

Comment:
```swift
movies = dto.data.movies.nowPlaying.edges.map(MoviePreviewDTOWrapper.init(dto:))
```

Reply:
> ```swift
> movies = dto.data.movies.nowPlaying.edges.map(MoviePreviewDTOWrapper.init(dto:))
> ```

it doesn't work, cause i need exactly .node property of each Edges, thats why { ...$0.node } is used

Reply:
Ok, let's leave it like that. Or you could do it like:
```swift
movies = dto.data.movies.nowPlaying.edges.map(\.node).map(MoviePreviewDTOWrapper.init(dto:))
```